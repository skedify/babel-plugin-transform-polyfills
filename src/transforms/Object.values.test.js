import compile from '../../test/utils/compile'

describe('Object.values', () => {
  it('should replace Object.values with a polyfill', () => {
    const result = compile(
      `
      const c = Object.values({ a: 1, b: 2 })
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const e = Object.values({ a: 1, b: 2 })
      const f = Object.values({ a: 1, b: 2 })
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const c = Object.keys(Object.values({ a: 1, b: 2 }))
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when using apply', () => {
    const result = compile(
      `
      const c = Object.values.apply(null, [{ a: 1, b: 2 }])
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when using call', () => {
    const result = compile(
      `
      const c = Object.values.call(null, { a: 1, b: 2 })
    `
    )

    expect(result).toMatchSnapshot()
  })
})
