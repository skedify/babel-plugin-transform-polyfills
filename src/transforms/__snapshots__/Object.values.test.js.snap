// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Object.values should not import polyfill more than once 1`] = `
"import _Object_values from \\"babel-plugin-transform-polyfills/polyfills/Object/values\\";

const e = _Object_values({
  a: 1,
  b: 2
});

const f = _Object_values({
  a: 1,
  b: 2
});"
`;

exports[`Object.values should replace Object.values with a polyfill 1`] = `
"import _Object_values from \\"babel-plugin-transform-polyfills/polyfills/Object/values\\";

const c = _Object_values({
  a: 1,
  b: 2
});"
`;

exports[`Object.values should work when part of a call chain 1`] = `
"import _Object_values from \\"babel-plugin-transform-polyfills/polyfills/Object/values\\";
const c = Object.keys(_Object_values({
  a: 1,
  b: 2
}));"
`;

exports[`Object.values should work when using apply 1`] = `
"import _Object_values from \\"babel-plugin-transform-polyfills/polyfills/Object/values\\";

const c = _Object_values.apply(null, [{
  a: 1,
  b: 2
}]);"
`;

exports[`Object.values should work when using call 1`] = `
"import _Object_values from \\"babel-plugin-transform-polyfills/polyfills/Object/values\\";

const c = _Object_values.call(null, {
  a: 1,
  b: 2
});"
`;
